### PR TITLE
Hiero: Effect Categories - OP-7397

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
This PR introduces `Effect Categories` for the Hiero settings. This allows studios to split effect stacks into meaningful subsets.

## Testing notes:
1. Setup `Effect Categories` in Hiero addon settings (see documentation).
2. Setup a clip in Hiero with effects corresponding to the settings.
3. Publish clip and effects.
4. Load into Nuke and verify the effects are grouped correctly.

### CODE REVIEW

**Comment by tokejepsen** [comment]
Docs https://github.com/ynput/ayon-documentation/pull/123

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
awesome work! Can we also add default values as you were showing in documentation into the variable `DEFAULT_PUBLISH_PLUGIN_SETTINGS`?

**Comment by tokejepsen** [comment]
> Can we also add default values as you were showing in documentation into the variable DEFAULT_PUBLISH_PLUGIN_SETTINGS?

I would maybe argue that we should not have any defaults to maintain backwards compatibility?

**Comment by tokejepsen** [comment]
> It is not collecting soft effect nodes and then those are not serialized into json files.

@jakubjezek001 I'm not entirely sure what you meant here, but I've fixed the case where effects are not categorized, in which case we just dump the effects into the "main" effects.

File: openpype/hosts/hiero/plugins/publish/collect_clip_effects.py [hunk 1]
```diff
@@ -70,29 +72,62 @@ def process(self, instance):
...
-        data["effects"] = effects
 
-        # create new instance
-        _instance = instance.context.create_instance(**data)
-        self.log.info("Created instance `{}`".format(_instance))
-        self.log.debug("instance.data `{}`".format(_instance.data))
+            # Some classes can have a number in them. Like Text2.
+            found_cls = ""
+            for cls in category_by_effect.keys():
+                if value["class"].startswith(cls):
```
Comment (jakubjezek001) [inline openpype/hosts/hiero/plugins/publish/collect_clip_effects.py:93]:
```suggestion
                if cls in value["class"]:
```

**Review by jakubjezek001 — APPROVED** [review]
Once the suggestion is implemented anyhow, then it is good for merge.

**Pull request merged**